test/integration/consul-container/upgrade/healthcheck_test.go

```diff
@@ -0,0 +1,328 @@
+package consul_container
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+
+	consulCluster "github.com/hashicorp/consul/integration/consul-container/libs/consul-cluster"
+	consulNode "github.com/hashicorp/consul/integration/consul-container/libs/consul-node"
+
+	"github.com/hashicorp/consul/integration/consul-container/libs/utils"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+
+	"github.com/stretchr/testify/require"
+)
+
+var curImage = flag.String("uut-version", "local", "docker image to be used as UUT (unit under test)")
+var latestImage = flag.String("latest-version", "latest", "docker image to be used as latest")
+
+const retryTimeout = 10 * time.Second
+const retryFrequency = 500 * time.Millisecond
+
+// Test health check GRPC call using Current Clients and Latest GA Servers
+func TestLatestGAServersWithCurrentClients(t *testing.T) {
```

Comment:
Technically speaking this scenario isn't supported is it.

Server versions must always be greater than the client versions. Maybe I am reading this wrong but it sounds like this would be used to test maybe a 1.11.5 servers with 1.11.6 clients (assuming 1.11.6 hasn't been released yet)

Reply:
I did not know that the requirements are to always have servers upgraded first. Good to know I will remove this scenario.